src/core/src/main/java/org/apache/jmeter/testelement/TestPlan.java

```diff
@@ -82,12 +85,23 @@ public void prepareForPreCompile()
     }
 
     /**
-     * Fetches the functional mode property
-     *
+     * Fetches the functional mode property<br>
+     * Could be change for no-GUI test with jmeter property: {@code PROP_FUNCTIONAL_MODE}
      * @return functional mode
      */
     public boolean isFunctionalMode() {
-        return getPropertyAsBoolean(FUNCTIONAL_MODE);
+        boolean functionalModeDefault = getPropertyAsBoolean(FUNCTIONAL_MODE);
+        log.debug("functionalModeDefault=" + functionalModeDefault);
+        boolean functionalModeReturn = functionalModeDefault;
+        if (isNonGui()) {
+            String propFunctionalModeProperty = JMeterUtils.getProperty(PROP_FUNCTIONAL_MODE);
+            if (propFunctionalModeProperty != null) {
+                functionalModeReturn = JMeterUtils.getPropDefault(PROP_FUNCTIONAL_MODE, functionalModeDefault);
+                log.info("Change with property " + PROP_FUNCTIONAL_MODE + ", value=" + functionalModeReturn);
```

Comment:
This will be visible to the end users.  What does `Change with property` mean?

Reply:
_>What does Change with property mean?__
Change with external property when you launch JMeter cli likes :
jmeter.bat -n -Jjmeter.test_plan.functional_mode=false

Reply:
Changed with proposal modifications log message for end user

Reply:
If you go with `jmeter.test_plan.tearDown_on_shutdown=true` (remove ` value`), then it would be understandable, and the users could search for that exact pattern everywhere. WDYT?